### DISCUSSION TITLE
BaramMesh not recognizing multiple regions

### DISCUSSION THREAD

**Discussion by MarcoWheemo** [nextfoam/baram#52] [Q&A]
Hey to all 

I am exploring Baram as an alternative CFD solution right now and I am quite impressed how easy it is to use! 

I am facing an issue with BaramMesh, where the two different regions are not recognized. The problem only arises with slightly more complex geometries. To give you an example.

1. First I tried a very simple CHT with two cylinders, one being the tube, the inner one being the fluid. After selecting the solid and the fluid region resp. it recognized the two regions and it created a good multi region mesh.

2. Now with a slightly more complex mesh baram puts all mesh elements in both regions (e.g. cellZoneSet solid now size 88902 & cellZoneSet fluid now size 88902, 88902 is the full number of elements). The geometry explained is a disk (solid) with a flat cooling channel on one flat side of the disc going on round around it (the fluid should cool the surface, because heat is generated from the otherside of the disc). I have defined my stl in the following way. 
- interface/wall --> is the wall of the fluid and resp. of the solid
- inlet --> fluid inlet 
- outlet --> fluid outlet
- heat-sur --> the surface for the heat flux BC
- solid --> everything except interface/wall, inlet, outlet and heat-sur

As I was able to successfully create a multi-region mesh in case 1, I really don't know what I am doing wrong here. 

PS: Is there a way to go to the scripts and define the regions manually to circumvent the issue I am having? 

Edit: I am using Windows and Baram23.3.5

![image](https://github.com/nextfoam/baram/assets/124031742/87815a76-f33b-48b7-8614-0516c0d4fc7c)

![image](https://github.com/nextfoam/baram/assets/124031742/90633804-5fc6-452d-a6f9-381531707e8c)

Thanks a lot in advance.

**Comment by kdg5424** [comment]
Hi @MarcoWheemo. I'm Donggyu Kim.
First of all, thank you for your attention in BARAM.

Could you tell me how did you select the boundary type of Geometry?
Just capture the step '**1. Geometry**'.
I want to know what boundary type you selected.

**Comment by MarcoWheemo** [comment]
Hi Donggyu, I am Marco

Thanks for the super fast reply!
![image](https://github.com/nextfoam/baram/assets/124031742/dc070885-5bba-47c4-879e-fcacd05089bd)
This would be the first step. 

Let me know if you need more information.

**Reply by kdg5424** [reply]
Well... it looks like there is no problem with your settings.
Can you send me your stl files?
Let me try with your stl file.

dkkim@nextfoam.co.kr

It's my email link.

**Reply by MarcoWheemo** [reply]
I was able to test a bit more and I found something weird. 

To test different setups, I thought of creating a similar geometry directly in Gmsh and then exporting it into an STL and this way it worked. But with the .step file from my cad software then converting to a STL something gets lost. 

I thought it would be good to share

**Comment by MarcoWheemo** [comment]
Just sent it out. Thanks for your support!

**Comment by MarcoWheemo** [comment]
Is there a way to convert the mesh created by openfoam itself using the splitMeshRegions utility to a mesh that is supported by baram?

**Comment by arniendf** [comment]
Hi Marco,
I am having similar issues regarding BaramMesh auto selection of boundary surfaces in slightly more complex geometry.

Did you ever get resolution on this challenge ?

**Comment by MarcoWheemo** [comment]
Hi arniendf

Unfortunately not, but for me this was on an older version. I had the issue mainly when I tried to come from a stl file. My process now is a bit more complex, but it is much more reliable. 

1. In a CAD Program select the surfaces that are of a specific region you want to select. E.g. the outside of the solid
2. Export each region of interest as a .step file
3. Import the .step file in to gmsh and make a mesh out of it. The finer the mesh the better the .stl. Then export it as an output. 
4. Now you should have a selection of different .stl files, which when put together make up your entire geometry. 
5. Import all of your .stl files into baramMesh. This will help. 

It is very very important that you have good .stl files, with no holes or leaks, otherwise you will always run into issues. 

For me I tend to simplify the geometry, such that I can reduce the number of surfaces.

**Reply by SergioP1975** [reply]
Hi @MarcoWheemo , did you try increasing the resolution of the exported STL in the CAD program?

**Reply by MarcoWheemo** [reply]
It really helps if you have fine meshes on the surface in gmsh before you export an .stl file. This will have just a better representation, the reason for doing it in gmsh is because I can control the mesh very easily.

**Comment by MateuszJakubowski** [comment]
Hello, I have similar issue. Baram creates mesh only for single region. When I move the point of region to another volume it meshes the other one. 

In the snappyHexMesh file there is only one `locationInMesh` point. 

<img width="1844" height="1427" alt="image" src="https://github.com/user-attachments/assets/dddce28b-ceb4-4775-9fac-5f4e5034695b" />

<img width="1844" height="1427" alt="image" src="https://github.com/user-attachments/assets/41e2da06-0b4b-453b-b6cd-3c3f3e0bba8c" />

**Reply by jiban** [reply]
BARAM Mesh splits regions with `splitMeshRegions` command during export.
And that is the reason you could see `locationInMesh` with only one point.
Please let me know if you couldn't get multi-region after export.

-- Jake

**Reply by MateuszJakubowski** [reply]
Okay, thanks for clarification.
I did not export the mesh yet, since `internalMesh` appears only in one region. Do I do something wrong?
I can provide my .bf case folder if that helps.

<img width="1553" height="933" alt="image" src="https://github.com/user-attachments/assets/7ea42ac7-494e-4934-b7a4-50397c65365a" />

<img width="1323" height="797" alt="image" src="https://github.com/user-attachments/assets/b0e2ebe3-98fc-4431-ade8-6f8e5ffbe352" />

<img width="1320" height="803" alt="image" src="https://github.com/user-attachments/assets/43261e64-2feb-408a-9d29-18ad5728d51b" />

**Reply by jiban** [reply]
Sure. I will check the case if you share it here.
By the way, it would be `.bm` rather than `.bf` since it is a BARAM Mesh case.

**Reply by MateuszJakubowski** [reply]
You are right, I meant `.bm` folder

The file `.zip` exceeded 25 MB so I have to share it by google drive. 

Folder:
[https://drive.google.com/drive/folders/1Q43GgmuNncDaia2nnnqNmu6MR2meTOfw?usp=sharing](url)

or `.zip` file:
[https://drive.google.com/file/d/14PnDFiCL1pWp2flLCb2H2cNmv9yFajqu/view?usp=drive_link](url)

**Reply by jiban** [reply]
You've shared `case` folder under `.bm` folder.
I need `.bm` folder itself.

**Reply by MateuszJakubowski** [reply]
Sorry, I've just uploaded everything into the folder link provided above

**Comment by jiban** [comment]
You seem to share different case rather than your problematic case.
It contains only one region point...

<img width="1710" height="627" alt="image" src="https://github.com/user-attachments/assets/9f646238-d7da-4a3a-9f34-18a20421a23d" />

**Reply by MateuszJakubowski** [reply]
ah, I was in rush and uploaded data from wrong folder, sorry!!

Would you mind taking another look?

the case should look as below

<img width="1318" height="783" alt="image" src="https://github.com/user-attachments/assets/9d06d431-543e-4a79-a803-1b9a8a6e7f31" />


<img width="1316" height="786" alt="image" src="https://github.com/user-attachments/assets/a6775e6a-9a8f-464b-a86f-dd07b952313a" />

**Reply by MateuszJakubowski** [reply]
folder link is the same: https://drive.google.com/drive/folders/1Q43GgmuNncDaia2nnnqNmu6MR2meTOfw?usp=sharing

**Reply by jiban** [reply]
It's because your two regions are closed.

<img width="1061" height="633" alt="image" src="https://github.com/user-attachments/assets/9f92865c-7574-4fb5-87e7-3257de87cee9" />

The two regions should be connected with only one surface of interface type.
Please refer to this post.
[Baram Mesh for multi-region mesh preparation](https://github.com/nextfoam/baram/discussions/60#discussioncomment-8759053)

Left *HotRegion* and righ *ColdRegion* are connected via *InterRegionWall*, and this is the only surface between the two regions.
The two regions are open to each other if this inter-region interface is not there.
I think it's tricky.
Unfortunately this is how `snappyHexMesh`, which is underlying BARAM Mesh, is working.

<img width="923" height="642" alt="image" src="https://github.com/user-attachments/assets/8244b200-a675-448e-9d1f-f43430d22bf1" />

**Reply by MateuszJakubowski** [reply]
After following the instruction work like a charm. 

Thank you for very responsive support!
